api/gameauth.js

```diff
@@ -0,0 +1,59 @@
+const gameauth = require("../util/gameauth.js");
+const events = require("../util/events.js");
+const api_users = require("./users.js");
+
+/**
+ * Handles `/api/gameauth/` endpoint requests. This endpoint supports the following commands:
+ *
+ * - `connect`: Binds the provided auth code to the user and waits for it to be verified before upgrading the request to a WebSocket connection
+ * - `verify`: Checks the bound code and resolves the promise of a WebSocket connection
+ *
+ * @param {[string, number]} args The arguments for the API request, in order: command name, numeric auth code, SteamID (only for game client)
+ * @param {HttpRequest} request The HTTP request object
+ * @returns {Promise<string|undefined>} The response of the API request. In the event of <undefined>, the Request is upgraded to a WebSocket connection
+ */
+module.exports = async function (args, request) {
+
+  const [command, authcode, steamid] = args;
+
+  const server = epochtal.data.events.server;
+
+  switch (command) {
+
+    case "connect": {
+
+      // Ensure that a game client event for this user exists
+      const eventName = `game_${steamid}`;
+      try {
+        await events(["get", eventName]);
+      } catch {
+        await events(["create", eventName]);
+      }
+
+      // Stall here until the auth code is verified by the logged in user
+      await new Promise(async function (resolve) {
+        await gameauth(["set", steamid, authcode, resolve]);
+      });
```

Comment:
t- there's gotta be some cleaner syntax for this

Reply:
i honestly don't see anything wrong with this?

Reply:
i mean optimally it's just a function call, putting the promise block around it makes it much harder to read.
Besides you don't actually use the resolve

Reply:
we do use the resolve though? that's the whole point? it's passed into the `gameauth` call as a callback function, which is then called after auth is verified

Reply:
ohhh I understand, so it's waiting for the auth verification which will callback to resolve and *that* is what you're awaiting.. my bad!

Reply:
yes! stalling the actual util call would be _very_ silly. instead, the `await`ed `gameauth` call here returns almost instantly with either "SUCCESS" or some error, as it should, and as most utils do. the wrapping promise is indeed what performs the actual waiting. we can't make this fully async either, because that would mean returning out of the api call, which means sending a response to the http request, after which you obviously can't elevate the connection to a websocket anymore.